src/openrct2-ui/windows/Guest.cpp

```diff
@@ -1982,7 +1982,7 @@ void window_guest_inventory_paint(rct_window* w, rct_drawpixelinfo* dpi)
     gfx_draw_string_left(dpi, STR_CARRYING, nullptr, COLOUR_BLACK, screenCoords);
     screenCoords.y += 10;
 
-    for (ShopItem item = ShopItem::Balloon; item < ShopItem::Count; item++)
+    for (ShopItem item = ShopItem::Balloon; item < ShopItem::Count; item = ShopItem(EnumValue(item) + 1))
```

Comment:
Is it not possible to implement `++` for ShopItem?

Reply:
I tend to not like having ++ for enum classes as you can get values that are not valid. For example this loop will include a couple invalid items as not all 64 bits are in use.

Reply:
Well, exactly - this has the same downside, except it looks very ugly on top.

Can we define a ++ that skips these invalid values?